File: Python/4_Pre_Processing/Data_Imputation/DataImputation.py
#########################################################################################################
# Name             : Data Imputation
# Date             : 04-21-2015
# Author           : Christopher M
# Dept             : BEI Business Analytics
#########################################################################################################
# ver    user        date(YYYYMMDD)        change  
# 1.0    CMooney      20150421             initial
#########################################################################################################


def ValueImputationMedian(frame,Target,IndicatorCol=False):
    if IndicatorCol == True:
        Col = Target + "_Missing"
        frame[Col] = frame[Target].isnull()
    
    frame[Target] = frame[Target].fillna(frame[Target].median())
    
    return(frame)
    
def ValueImputationAverage(df,Target,IndicatorCol=False):
    if IndicatorCol == True:
        Col = Target + "_Missing"
        df[Col] = df[Target].isnull()
    
    df[Target] = df[Target].fillna(df[Target].mean())
    
    return(df)    

File: R/4_Pre_Processing/Data_Imputation/Data_Imputation.r
###############################################################################################################################
# Name             : Data_Imputation
# Date             : 04-12-2015
# Author           : Christopher M
# Dept             : Business Analytics
# Purpose          : Example of how to impute values (mean,median)
# Called by        : Not in production
###############################################################################################################################
# ver    user        date(YYYYMMDD)        change  
# 1.0    w47593      20150412             initial
###############################################################################################################################
 
##_____________________________________________________________________________________________________________________________
# Example of how to impute missing values with the mean and median

# remove airquality incase you previously changed the structure of the dataframe
rm(airquality)
airquality[is.na(airquality)] <- median(airquality$Ozone,na.rm=TRUE)

rm(airquality)
airquality[is.na(airquality)] <- mean(airquality$Ozone,na.rm=TRUE)